auth/src/VSCodeAzureSubscriptionProvider.ts

```diff
@@ -0,0 +1,241 @@
+    }
+
+    /**
+     * Gets the subscription filters that are configured in `azureResourceGroups.selectedSubscriptions`. To
+     * override the settings with a custom filter, implement a child class with `getSubscriptionFilters()`
+     * and/or `getTenantFilters()` overridden.
+     *
+     * If no values are returned by `getSubscriptionFilters()`, then all subscriptions will be returned.
+     *
+     * @returns A list of subscription IDs that are configured in `azureResourceGroups.selectedSubscriptions`.
+     */
+    protected async getSubscriptionFilters(): Promise<SubscriptionId[]> {
+        const config = vscode.workspace.getConfiguration('azureResourceGroups');
+        const fullSubscriptionIds = config.get<SubscriptionId[]>('selectedSubscriptions', []);
+        return fullSubscriptionIds.map(id => id.split('/')[1]);
+    }
+
+    /**
+     * Gets the tenants available to a user.
+     *
+     * @returns The list of tenants visible to the user.
+     */
+    private async getTenants(): Promise<TenantIdDescription[]> {
+        const { client } = await this.getSubscriptionClient('organizations');
+        const tenants: TenantIdDescription[] = [];
+
+        for await (const tenant of client.tenants.list()) {
+            tenants.push(tenant);
+        }
+
+        return tenants;
+    }
+
+    /**
+     * Gets the subscriptions for a given tenant.
+     *
+     * @param tenantId The tenant ID to get subscriptions for.
+     *
+     * @returns The list of subscriptions for the tenant.
+     */
+    private async getSubscriptionsForTenant(tenantId: string): Promise<AzureSubscription[]> {
+        const { client, credential, authentication } = await this.getSubscriptionClient(tenantId);
+        const environment = getConfiguredAzureEnv();
+
+        const subscriptions: AzureSubscription[] = [];
+
+        for await (const subscription of client.subscriptions.list()) {
+            subscriptions.push({
+                authentication: authentication,
+                environment: environment,
+                credential: credential,
+                isCustomCloud: environment.isCustomCloud,
+                /* eslint-disable @typescript-eslint/no-non-null-assertion */
+                name: subscription.displayName!,
+                subscriptionId: subscription.subscriptionId!,
+                /* eslint-enable @typescript-eslint/no-non-null-assertion */
+                tenantId: tenantId,
+            });
+        }
+
+        return subscriptions;
+    }
+
+    /**
+     * Gets a fully-configured subscription client for a given tenant ID
+     *
+     * @param tenantId The tenant ID to get a client for
+     *
+     * @returns A client, the credential used by the client, and the authentication function
+     */
+    private async getSubscriptionClient(tenantId: 'organizations' | string): Promise<{ client: SubscriptionClient, credential: TokenCredential, authentication: AzureAuthentication }> {

```

Comment:
Is `'organizations'` representing the default tenant?

Reply:
If so, I'm not sure having a tenant id that represents the default tenant is really needed. If I understand what the default tenant is, it's the tenant associated with the current session, whatever tenant the user used when logging in.

The default tenant id can be grabbed from the session id. Can we just use that instead of `'organizations'`?

Reply:
`organizations` is the scope needed to get a list of _tenants_ that a user has access to. It can be more than one.

Reply:
> The default tenant id can be grabbed from the session id

We can but didn't we want to list all subs across all tenants?